arm9/lib/include/NNS_g2d.h

```diff
@@ -0,0 +1,264 @@
+    NNS_G2D_CHARACTERMAPING_1D_256,
+    NNS_G2D_CHARACTERMAPING_2D,
+    NNS_G2D_CHARACTERMAPING_MAX
+
+} NNSG2dCharacterDataMapingType;
+
+typedef struct NNSG2dImageAttr
+{
+    GXTexSizeS                      sizeS;          // Image size (Invalid data is set during 1D mapping.)
+    GXTexSizeT                      sizeT;          // Image size (Invalid data is set during 1D mapping.)
+
+    GXTexFmt                        fmt;            // Image format
+    BOOL                            bExtendedPlt;   // Whether or not to use the extended palette
+
+    GXTexPlttColor0                 plttUse;        // How to use palette number 0
+    GXOBJVRamModeChar               mappingType;    // Mapping mode
+}
+NNSG2dImageAttr;
+
+typedef struct NNSG2dVRamLocation
+{
+    u32                 baseAddrOfVram[ NNS_G2D_VRAM_TYPE_MAX ];
+}
+NNSG2dVRamLocation;
+
+typedef struct NNSG2dImageProxy
+{
+    NNSG2dVRamLocation        vramLocation;
+    NNSG2dImageAttr           attr;
+}
+NNSG2dImageProxy;
+
+typedef struct NNSG2dPaletteCompressInfo
+{
+    u16               numPalette;
+    u16               pad16;
+    void*             pPlttIdxTbl;        // offset addr.
+
+}
+NNSG2dPaletteCompressInfo;
+
+void NNS_G2dLoadImage1DMapping
+(
+    const NNSG2dCharacterData*  pSrcData,
+    u32                         baseAddr,
+    NNS_G2D_VRAM_TYPE           type,
+    NNSG2dImageProxy*           pImgProxy
+);
+
+void NNS_G2dLoadImage2DMapping
+(
+    const NNSG2dCharacterData*  pSrcData,
+    u32                         baseAddr,
+    NNS_G2D_VRAM_TYPE           type,
+    NNSG2dImageProxy*           pImgProxy
+);
+
+typedef struct NNSG2dImagePaletteProxy
+{
+
+    GXTexFmt                  fmt;          // palette format
+    // (Obtainable values are limited to GX_TEXFMT_PLTT16 and GX_TEXFMT_PLTT256)
+
+    BOOL                      bExtendedPlt; // Use expanded palette?
+
+    NNSG2dVRamLocation        vramLocation;
+}
+NNSG2dImagePaletteProxy;
+
+void NNS_G2dLoadPalette
+(
+    const NNSG2dPaletteData*    pSrcData,
+    u32                         addr,
+    NNS_G2D_VRAM_TYPE           type,
+    NNSG2dImagePaletteProxy*    pPltProxy
+);
+
+void NNS_G2dLoadPaletteEx
+(
+    const NNSG2dPaletteData*            pSrcData,
+    const NNSG2dPaletteCompressInfo*    pCmpInfo,
+    u32                                 addr,
+    NNS_G2D_VRAM_TYPE                   type,
+    NNSG2dImagePaletteProxy*            pPltProxy
+);
+
+typedef struct NNSG2dCellVramTransferData
+{
+    u32        srcDataOffset;
+    u32        szByte;
+
+}
+NNSG2dCellVramTransferData;
+
+typedef struct NNSG2dVramTransferData
+{
+    u32                                 szByteMax;             // Maximum byte count during all VRAM transfers
+    NNSG2dCellVramTransferData*         pCellTransferDataArray;// This is the length of the numCells amount of the maintained NNSG2dCellDataBank.
+    //
+
+}
+NNSG2dVramTransferData;
+
+typedef struct NNSG2dCellOAMAttrData
+{
+    u16         attr0;

```

Comment:
these are bitfields and probably should be treated as such (union with u16?)

Reply:
Possibly. For now, I'm keeping SDK headers as-is.